### PR TITLE
Move base and data encoders to data

### PR DESCRIPTION
**Pull request by tomvanmele**

### CODE REVIEW

**Comment by tomvanmele** [comment]
todo (or not)

* move json functions to data module
* add session object
* store compas data in Rhino object user data and vice versa
* add tutorial about data
* underscore the data property to prevent direct access
* change `item` in artist and object to `data`

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,10 +11,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-* Added `pretty` parameter to `compas.json_dump` and `compas.json_dumps`
+* Added `pretty` parameter to `compas.json_dump` and `compas.json_dumps`.
+* Added `compas.data.Data` as base object for all data objects (geometry, data structures, ...).
 
 ### Changed
 
+* Moved `compas.utilities.DataEncoder` to `compas.data`.
+* Moved `compas.utilities.DataDecoder` to `compas.data`.
+* Changed base object of `compas.datastructures.Datastructure` to `compas.data.Data`.
+* Changed base object of `compas.geometry.Primitive` to `compas.data.Data`.
```
Comment (gonzalocasas) [inline CHANGELOG.md:22]: Maybe add `Renamed Base to Data` or similiar?

File: src/compas/files/urdf.py [hunk 2]
```diff
@@ -322,7 +322,7 @@ def build_kwargs_by_type(cls, elements, parser_type, default_namespace):
         return result
 
 
-class URDFGenericElement(Base):
+class URDFGenericElement(Data):
```
Comment (gonzalocasas) [inline src/compas/files/urdf.py:325]: In this class (as in many others in the robot model implementation), the methods `to_json`/`from_json` could be removed now.

**Review by gonzalocasas — APPROVED** [review]
LGTM! I would remove the now redundant `from_json` and `to_json` from everywhere if possible.

**Pull request merged**